auxx/src/Command/Tx.hs

```diff
@@ -153,7 +153,7 @@ sendToAllGenesis sendActions (SendToAllGenesisParams duration conc delay_ sendMo
                                   addTxFailed tpsMVar
                                   logError (sformat ("Error: "%build%" while trying to send to "%shown) err neighbours)
                               Right (tx, _) -> do
-                                  res <- submitTxRaw (immediateConcurrentConversations sendActions neighbours) tx
+                                  res <- submitTxRaw diffusion tx
```

Comment:
Previously the logic was to send to `neighbours`, which is constructed inside this function (`sendToAllGenesis`) and was passed to `immediateConcurrentConversations`.
Now we are using `diffusion` instead which is passed to this function as an argument and is never modified inside this function. So it doesn't know anything about `neighbours` and will probably always send to `ccPeers`.
Currently `neighbours` are only for logging, but not really used by `submitTxRaw`.
How is it supposed to work? Is this logic broken?

Reply:
The diffusion layer takes care of choosing where to send data.

Reply:
I understand it. But there is `SendMode` type which specifies how transactions are sent to the network. And there is such code:
```
            neighbours <- case sendMode of
                SendNeighbours -> return ccPeers
                SendRoundRobin -> return [ccPeers !! (n `mod` nNeighbours)]
                SendRandom -> do
                    i <- liftIO $ randomRIO (0, nNeighbours - 1)
                    return [ccPeers !! i]
```

AFAIU, this code is effectively ignored currently. Is it correct?
If it's correct, how hard would it be to fix it? Is it even possible with current interface? If it's not possible, maybe it makes sense to revise the interface. If it's possible but hard, you should create an issue and write comments that `SendMode` is ignored. If we just merge it as is, someone can use benchmarks with different `SendMode`s and be surprised that `SendMode` is effectively ignored (and also messages are confusing).

P. S. Maybe I just don't get the code and it somehow magically works as it should.

Reply:
> If it's correct, how hard would it be to fix it? Is it even possible with current interface? If it's not possible, maybe it makes sense to revise the interface.

The point of this abstraction is that the logic doesn't deal with network topology. I'll remove `SendMode`, as I see no reason to have it. When I send a transaction, I want it to go out and permeate the network. Only the diffusion layer is capable of deciding how to make that happen, so there's nothing more to specify.

Reply:
I didn't work on benchmarks, so I don't know whether there is anybody who needs this `SendMode`. I personally don't need it, but it probably makes sense to ask in sl-benchmarks whether it's safe to delete this feature.